adapters/handlers/rest/clusterapi/indices_payloads.go

```diff
@@ -210,19 +210,25 @@ func (p objectListPayload) Unmarshal(in []byte) ([]*storobj.Object, error) {
 	r := bytes.NewReader(in)
 
 	for {
-		_, err := r.Read(reusableLengthBuf)
+		n, err := r.Read(reusableLengthBuf)
 		if errors.Is(err, io.EOF) {
 			break
 		}
 		if err != nil {
 			return nil, err
 		}
+		if n != len(reusableLengthBuf) {
```

Comment:
this may be solvable, if some buffered reader was used along the chain of readers it may just return fewer bytes but it might not mean it could not read more, it may be convenient to use `io.ReadFull` as it will do its best to read the entire requested content